Std/Lean/Json.lean

```diff
@@ -0,0 +1,35 @@
+/-
+ Copyright (c) 2022 E.W.Ayers. All rights reserved.
+ Released under Apache 2.0 license as described in the file LICENSE.
+ Authors: E.W.Ayers
+-/
+
+import Lean.Data.Json
+
+open Lean
+
+instance : OfScientific JsonNumber where
+  ofScientific mantissa exponentSign decimalExponent :=
+    if exponentSign then
+      {mantissa := mantissa, exponent := decimalExponent}
+    else
+      {mantissa := (mantissa * 10 ^ decimalExponent : Nat), exponent := 0}
+
+instance : Neg JsonNumber where
+  neg jn := ⟨- jn.mantissa, jn.exponent⟩
+
+instance : ToJson Float where
+  toJson x :=
+    if x == 0.0 then 0 else
+    let s := toString (if x < 0.0 then - x else x)
+    match Lean.Syntax.decodeScientificLitVal? <| s with
+    | none =>
+      match s with
+      | "inf" => "inf" -- [todo] emit a warning
+      | "-inf" => "-inf"
+      | "nan" => "nan"
+      | _ => panic! s!"unhandled float string {s}"
+    | some (m, e, de) =>
+      let j : JsonNumber := OfScientific.ofScientific m e de
+      let j := if x < 0.0 then -j else j
+      Json.num j
```

Comment:
This instance looks super hackish, translating stuff to strings and then back, and constructing invalid JSON literals like `nan`. What is going on here? (If you want to translate invalid floats I think the usual method is to use `null` instead.)